pdf/pdfrenderthread.cpp

```diff
@@ -33,6 +33,72 @@ class PDFRenderThreadQueue;
 
 const QEvent::Type Event_JobPending = QEvent::Type(QEvent::User + 1);
 
+class SearchThread: public QThread
+{
+    Q_OBJECT
+public:
+    SearchThread(Poppler::Document *document, QObject *parent = 0)
+        : QThread(parent), m_document(document)
+    {
+    }
+    ~SearchThread()
+    {
+        requestInterruption();
+        wait();
+    }
+
+    QList<QPair<int, QRectF>> m_matches;
+
+    void start(const QString& search, uint startPage = 0)
+    {
+        requestInterruption();
+        wait();
+
+        m_search = search;
+        m_startPage = startPage;
+        QThread::start();
+    }
+
+    void run() {
+        m_matches.clear();
+        for (int i = 0; i < m_document->numPages() && !isInterruptionRequested(); ++i) {
+            int ipage = (m_startPage + i) % m_document->numPages();
+            Poppler::Page *page = m_document->page(ipage);
+
+            double sLeft, sTop, sRight, sBottom;
+            float scaleW = 1.f / page->pageSizeF().width();
+            float scaleH = 1.f / page->pageSizeF().height();
+            bool found;
+            found = page->search(m_search, sLeft, sTop, sRight, sBottom,
+                                 Poppler::Page::FromTop,
+                                 Poppler::Page::IgnoreCase);
+            while (found) {
```

Comment:
Yes, I would like to add this functionality also, but I'm wondering currently how to simply signal from the search thread that some additional matches are available. Maybe by signaling the length of the m_matches list every ten pages or so and do the beginRowInsert() things in SearchModel from that simple integer information.

Let's see later !